phoenix-core/src/main/java/org/apache/phoenix/jdbc/PhoenixConnection.java

```diff
@@ -450,7 +463,7 @@ private void closeStatements() throws SQLException {
         // from modifying this list.
         this.statements = Lists.newArrayList();
         try {
-            mutationState.rollback(this);
+            mutationState.clear();
```

Comment:
I'm pretty sure we want to call mutationState.rollback() here instead of mutationState.clear(). Add a test like this to confirm:
- upsert a few rows to a transactional table
- query the table (which will write you upserts but not commit them)
- close the connection
I think the writes will get left hanging and once the tx times out, I'm not sure what'll happen. The txID may end up on the invalid list (which is not great) or not (which is even worse).